PWGHF/TableProducer/candidateSelectorDplusToPiKPi.cxx

```diff
@@ -161,9 +164,26 @@ struct HfCandidateSelectorDplusToPiKPi {
     if (std::abs(candidate.maxNormalisedDeltaIP()) > cuts->get(pTBin, "max normalized deltaIP")) {
       return false;
     }
+    if (!isSelectedCandidateDcaXY(candidate)) {
+      return false;
+    }
     return true;
   }
 
+  /// Single-track cuts
+  /// \param candidate is the Ds candidate
+  /// \return true if all the prongs pass the selections
+  template <typename T1>
+  bool isSelectedCandidateDcaXY(const T1& candidate)
+  {
+    if (isSelectedTrackDcaXY(binsPtTrack, cutsSingleTrack, candidate.ptProng0(), candidate.impactParameter0()) &&
+        isSelectedTrackDcaXY(binsPtTrack, cutsSingleTrack, candidate.ptProng1(), candidate.impactParameter1()) &&
+        isSelectedTrackDcaXY(binsPtTrack, cutsSingleTrack, candidate.ptProng2(), candidate.impactParameter2())) {
+      return true;
+    }
+    return false;
```

Comment:
You can return the `bool` value of the condition directly if you don't plan to add other checks. If you do, plan to add other checks, it is better to invert the logic and return `false` as soon as one check fails and return `true` at the end otherwise.
```suggestion
    return (isSelectedTrackDcaXY(binsPtTrack, cutsSingleTrack, candidate.ptProng0(), candidate.impactParameter0()) &&
        isSelectedTrackDcaXY(binsPtTrack, cutsSingleTrack, candidate.ptProng1(), candidate.impactParameter1()) &&
        isSelectedTrackDcaXY(binsPtTrack, cutsSingleTrack, candidate.ptProng2(), candidate.impactParameter2()));
```

Reply:
Yes, indeed, so far I  want to check the single-track DCA, hence I can directly return the bool value of the condition. I will implement this suggestion and update the PR. Thanks a lot!